### ISSUE TITLE
Potential bug in receiver.detach()

### ISSUE THREAD

**Issue by ulisesbocchio** [amqp/rhea#354]
According to the documentation one could do `receiver.detach()` to detach a link without closing it. Once invoked, the only difference that I see with `receiver.close()` is the following:
```js
this.local.detach.closed = false;
```
vs `close` setting that same flag to `true`. The problem is, on `link.on_detach`, which is invoked after the receiver is detached, there's this:
```js
        process.nextTick(function () {
            if (self.state.marker === token) {
                self.close();
                process.nextTick(function () { self.remove(); });
            }
        });
```
which renders the receiver unresponsive to successive calls to `receiver.attach()`.
I tried adding the following statement before `process.nextTick` to avoid the receiver's close/removal
```js
if (!this.local.detach.closed) return
```
And after that I can successfully call `receiver.attach()` and reconnect the receiver after `receiver.detach()`, but then I start getting flooded with messages after adding 1 credit, and this error shows up:
```
Received transfer when credit was 0
```

Any ideas?

Thanks in advance!
Best,
Uli

**Comment by grs** [comment]
At present you would have to create a new receiver with the same link name (and most likely the same source). Directly calling attach on a closed receiver object is not currently supported. However from an AMQP perspective, opening a new receiver object should be equivalent.

**Comment by ulisesbocchio** [comment]
yeah, that's what I'm doing at the moment... since detach effectively closes the receiver, I created my own attach method that  recreates the receiver with the same name. I don't understand however what you mean by "and most likely the same source". Would you mind expanding on that?

Thanks!

**Comment by grs** [comment]
Probably not relevant. I think AMQP allows you to resume a link with a slightly different source configuration. In general though I think using the same source is more common.
